src/c/types/basic.c

```diff
@@ -57,7 +57,55 @@
     size_t alignment = ucdr_buffer_alignment(ub, SIZE); \
     ub->iterator += alignment; \
     ub->offset += alignment; \
-    if (ucdr_check_final_buffer_behavior(ub, SIZE)) \
+    if (ucdr_check_buffer_available_for(ub, SIZE)) \
+    { \
+        if (UCDR_MACHINE_ENDIANNESS == ENDIAN) \
+        { \
+            memcpy(ub->iterator, (void*)&value, SIZE); \
+        } \
+        else \
+        { \
+            UCDR_SERIALIZE_BYTE_ ## SIZE ## _CORE() \
+        } \
+        ub->iterator += SIZE; \
+        ub->offset += SIZE; \
+        ub->last_data_size = SIZE; \
+    } \
+    else if (ub->final > ub->iterator){ \
+        void* first_slot = ub->iterator; \
+        size_t first_slot_size = (size_t) (ub->final - ub->iterator); \
+        ub->iterator += first_slot_size; \
+        ub->offset += first_slot_size; \
+        if (ucdr_check_final_buffer_behavior(ub, SIZE - first_slot_size)) \
+        { \
+            if (UCDR_MACHINE_ENDIANNESS == ENDIAN) \
+            { \
+                memcpy(first_slot, (void*)&value, first_slot_size); \
+                memcpy(ub->iterator, ((uint8_t*)&value) + first_slot_size, SIZE - first_slot_size); \
```

Comment:
Are we moving `ub->iterator` out of the `[ub->init, ub->final]` range?

Reply:
Check here: https://github.com/eProsima/Micro-CDR/blob/e65a03ed0e5473dac2e488c9a1758b5fe4fef2a2/src/c/types/basic.c#L58